Allow users to override the extension passed to vale checker

krisbalintona opened the pull request:
Addresses #42.

Exposes an option, `flymake-collection-vale-extension-function`, that is a function that returns the extension (value of the --ext flag passed to vale) of the current file. The default value, the function
`flymake-collection-vale-default-extension-function`, preserves the previous behavior of the vale checker.

Please let me know if any changes are in order.

krisbalintona committed:
feat: Allow users to override the extension passed to vale checker

Exposes an option, `flymake-collection-vale-extension-function`, that is a
function that returns the extension (value of the --ext flag passed to vale) of
the current file. The default value, the function
`flymake-collection-vale-default-extension-function`, preserves the previous
behavior of the vale checker.

krisbalintona committed:
Merge branch 'mohkale:release' into vale-extension-function

krisbalintona commented:
@mohkale Hello,

Just wanted to bump this PR. No rush.

mohkale commented:
That's weird, the CI isn't running here. Pushing a dummy commit to see if that triggers it.

mohkale committed:
Merge remote-tracking branch 'origin/release' into vale-extension-function

mohkale committed:
chore: Add workflow_dispatch option

mohkale commented:
Opened https://github.com/orgs/community/discussions/144852

krisbalintona commented:
Hi, just wanted to check if anything came from your PR regarding the CI not running?

mohkale commented:
Sorry, I have no idea why github isn't running the actions and I don't really have the bandwidth to investigate fully. Github actions configuration interface is kinda of a hot mess. Could you please use `make lint` and `make docker DOCKER_FLAGS='' DOCKER_COMMAND='make test'` to validate on your end. I tested this now and got:

```text
$ make lint
In flymake-collection-vale-default-extension-function:
src/checkers/flymake-collection-vale.el:40:39: Warning: reference to free variable ‘flymake-collection-source’
make: *** [Makefile:40: bin/checkers/flymake-collection-vale.elc] Error 1
```

krisbalintona committed:
fix: Pass flymake-collection-source as an argument

Previously flymake-collection-vale-default-extension-function tried to access
the value of flymake-collection-source directly. This is erroneous since that
variable is only accessible within the form defining the checker.

To fix this, we let flymake-collection-vale-default-extension-function accept a
buffer as an argument and call flymake-collection-vale-extension-function with
flymake-collection-source as the argument.

krisbalintona commented:
Hi @mohkale,

Sorry about that. My latest commit should do the trick.

mohkale commented:
Apologies for the late response. Thanks for contributing.

krisbalintona commented:
@mohkale It's no problem. Happy Year Years!
